### PR TITLE
default spec.spotMaxPrice in AzureManagedMachinePool

### LINKED ISSUE #4112: [AzureManagedCluster][Spot] `AzureManagedMachinePool` fluctuating between `Running` and `Provisioned` states
/kind bug

`AzureManagedMachinePool` in `AzureManagedCluster` with Spot enabled are experiencing continuous fluctuations between the `Running` and `Provisioned` states.

**What steps did you take and what happened:**
Deploy `AzureManagedCluster` with Provider Version 1.11.3

Below is the configuration of one of the affected `AzureManagedMachinePools`:
```yaml
- apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
  kind: AzureManagedMachinePool
  metadata:
    finalizers:
    - azurecluster.infrastructure.cluster.x-k8s.io
    labels:
      azuremanagedmachinepool.infrastructure.cluster.x-k8s.io/agentpoolmode: User
      cluster.x-k8s.io/cluster-name: esierra-dev
      clusterctl.cluster.x-k8s.io/move-hierarchy: "true"
    name: worker2-mp-2
    namespace: cluster-esierra-dev
    ownerReferences:
    - apiVersion: cluster.x-k8s.io/v1beta1
      blockOwnerDeletion: true
      controller: true
      kind: MachinePool
      name: worker2-mp-2
    resourceVersion: "12913"
  spec:
    availabilityZones:
    - "3"
    linuxOSConfig:
      sysctls:
        vmMaxMapCount: 262144
    mode: User
    name: worker2mp2
    nodeLabels:
      backup: "false"
    osDiskSizeGB: 50
    osDiskType: Managed
    osType: Linux
    providerIDList:
    - azure:///subscriptions/REDACTED/resourceGroups/esierra-dev-nodes/providers/Microsoft.Compute/virtualMachineScaleSets/aks-worker2mp2-21099784-vmss/virtualMachines/0
    scaleDownMode: Delete
    scaleSetPriority: Spot
    scaling:
      maxSize: 6
      minSize: 1
    sku: Standard_D8_v3
    subnetName: esierra-dev
```

**What did you expect to happen:**
`AzureManagedMachinePool` remains in the `Running` state when it successfully passes `MachineHealthCheck` flights.

**Anything else you would like to add:**
Some of the pieces of evidence are:

 - Changes in the state of the MachinePool:
```bash
cluster-esierra-dev   worker2-mp-0   esierra-dev   1          Running       28m   v1.27.3
cluster-esierra-dev   worker2-mp-1   esierra-dev   1          Provisioned   28m   v1.27.3
cluster-esierra-dev   worker2-mp-2   esierra-dev   1          Provisioned   28m   v1.27.3
```
```bash
cluster-esierra-dev   worker2-mp-0   esierra-dev   1          Provisioned   28m   v1.27.3
cluster-esierra-dev   worker2-mp-1   esierra-dev   1          Running       28m   v1.27.3
cluster-esierra-dev   worker2-mp-2   esierra-dev   1          Provisioned   28m   v1.27.3
```
- Activity log screenshot:
image
Additional information in there:
```
Resource
/subscriptions/REDACTED/resourceGroups/esierra-dev/providers/Microsoft.ContainerService/managedClusters/esierra-dev/agentPools/worker2mp1
Operation name
Create or Update Agent Pool
Time stamp
Tue Oct 10 2023 12:03:39 GMT+0200 (Central European Summer Time)
Event initiated by
cloud-provisioner
```
- Some logs related to this in `capz-controller-manager`:
```bash
I1010 10:14:55.557724       1 reconciler.go:108] controllers.reconciler.Reconcile "msg"="processing" "AzureManagedMachinePool"={"name":"worker2-mp-0","namespace":"cluster-esierra-dev"} "controller"="azuremanagedmachinepool" "controllerGroup"="infrastructure.cluster.x-k8s.io" "controllerKind"="AzureManagedMachinePool" "name"="worker2-mp-0" "namespace"="cluster-esierra-dev" "reconcileID"="cb580257-6727-4ed2-b50c-b647170b0928" "request"="cluster-esierra-dev/worker2-mp-0" "x-ms-correlation-request-id"="5e07e639-5a94-402d-bf74-c9dd663727a9"
I1010 10:14:55.612367       1 azuremanagedmachinepool_controller.go:250] controllers.AzureManagedMachinePoolReconciler.reconcileNormal "msg"="Reconciling AzureManagedMachinePool" "AzureManagedMachinePool"={"name":"worker2-mp-0","namespace":"cluster-esierra-dev"} "controller"="azuremanagedmachinepool" "controllerGroup"="infrastructure.cluster.x-k8s.io" "controllerKind"="AzureManagedMachinePool" "name"="worker2-mp-0" "namespace"="cluster-esierra-dev" "reconcileID"="cb580257-6727-4ed2-b50c-b647170b0928" "x-ms-correlation-request-id"="5e07e639-5a94-402d-bf74-c9dd663727a9"
I1010 10:14:55.612885       1 azuremanagedmachinepool_reconciler.go:114] controllers.azureManagedMachinePoolService.Reconcile "msg"="reconciling managed machine pool" "AzureManagedMachinePool"={"name":"worker2-mp-0","namespace":"cluster-esierra-dev"} "controller"="azuremanagedmachinepool" "controllerGroup"="infrastructure.cluster.x-k8s.io" "controllerKind"="AzureManagedMachinePool" "name"="worker2-mp-0" "namespace"="cluster-esierra-dev" "reconcileID"="cb580257-6727-4ed2-b50c-b647170b0928" "x-ms-correlation-request-id"="5e07e639-5a94-402d-bf74-c9dd663727a9"
I1010 10:14:55.612991       1 client.go:83] agentpools.azureClient.CreateOrUpdate "msg"="sending request" "AzureManagedMachinePool"={"name":"worker2-mp-0","namespace":"cluster-esierra-dev"} "controller"="azuremanagedmachinepool" "controllerGroup"="infrastructure.cluster.x-k8s.io" "controllerKind"="AzureManagedMachinePool" "name"="worker2-mp-0" "namespace"="cluster-esierra-dev" "reconcileID"="cb580257-6727-4ed2-b50c-b647170b0928" "resumeToken"="{\"type\":\"AgentPoolsClientCreateOrUpdateResponse\",\"token\":{\"asyncURL\":\"https://management.azure.com/subscriptions/REDACTED/providers/Microsoft.ContainerService/locations/westeurope/operations/2d596d46-5e42-48a1-a837-b132def4a818?api-version=2017-08-31\\u0026t=638325296493969976\\u0026c=REDACTED\",\"locURL\":\"\",\"origURL\":\"https://management.azure.com/subscriptions/REDACTED/resourceGroups/esierra-dev/providers/Microsoft.ContainerService/managedClusters/esierra-dev/agentPools/worker2mp0?api-version=2023-07-01\",\"method\":\"PUT\",\"finalState\":\"\",\"state\":\"InProgress\"}}" "x-ms-correlation-request-id"="5e07e639-5a94-402d-bf74-c9dd663727a9"
I1010 10:14:57.614416       1 azuremanagedmachinepool_controller.go:283] controllers.AzureManagedMachinePoolReconciler.reconcileNormal "msg"="requeuing due to transient failure" "AzureManagedMachinePool"={"name":"worker2-mp-0","namespace":"cluster-esierra-dev"} "controller"="azuremanagedmachinepool" "controllerGroup"="infrastructure.cluster.x-k8s.io" "controllerKind"="AzureManagedMachinePool" "error"="failed to reconcile machine pool worker2mp0: operation type PUT on Azure resource esierra-dev/worker2mp0 is not done. Object will be requeued after 15s" "name"="worker2-mp-0" "namespace"="cluster-esierra-dev" "reconcileID"="cb580257-6727-4ed2-b50c-b647170b0928" "x-ms-correlation-request-id"="5e07e639-5a94-402d-bf74-c9dd663727a9"
I1010 10:14:57.615424       1 reconciler.go:115] controllers.reconciler.Reconcile "msg"="successful" "AzureManagedMachinePool"={"name":"worker2-mp-0","namespace":"cluster-esierra-dev"} "controller"="azuremanagedmachinepool" "controllerGroup"="infrastructure.cluster.x-k8s.io" "controllerKind"="AzureManagedMachinePool" "name"="worker2-mp-0" "namespace"="cluster-esierra-dev" "reconcileID"="cb580257-6727-4ed2-b50c-b647170b0928" "request"="cluster-esierra-dev/worker2-mp-0" "x-ms-correlation-request-id"="5e07e639-5a94-402d-bf74-c9dd663727a9"
```

**Environment:**

- cluster-api-provider-azure version: 1.11.3
- Kubernetes version: (use `kubectl version`):  v1.27.0

### PR DESCRIPTION
**Pull request by nojnhuh**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:

This PR fixes the AzureManagedMachinePool controller from futilely endlessly reconciling based on a diff when no `spec.spotMaxPrice` is defined for CAPZ and the value for the AKS resource in Azure gets defaulted to -1. This PR adds logic to the defaulting webhook that will mirror the same defaulting done by AKS for Spot node pools.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #4112

**Special notes for your reviewer**:

- [X] cherry-pick candidate <- to 1.11, this feature does not exist in 1.10

**TODOs**:

- [X] squashed commits
- [ ] includes documentation
- [X] adds unit tests

**Release note**:

```release-note
Fixed a bug that caused AzureManagedMachinePools to endlessly reconcile Spot node pools when no `spec.spotMaxPrice` is set
```

### CODE REVIEW

**Review by jackfrancis — APPROVED** [review]
/lgtm

File: api/v1beta1/azuremanagedmachinepool_webhook.go [hunk 1]
```diff
@@ -80,6 +81,10 @@ func (mw *azureManagedMachinePoolWebhook) Default(ctx context.Context, obj runti
 		m.Spec.OSType = ptr.To(DefaultOSType)
 	}
 
+	if ptr.Deref(m.Spec.ScaleSetPriority, "") == "Spot" && m.Spec.SpotMaxPrice == nil {
```
Comment (CecileRobertMichon) [inline api/v1beta1/azuremanagedmachinepool_webhook.go:84]: do we have a const in the API for "Spot"?

Comment (nojnhuh) [reply api/v1beta1/azuremanagedmachinepool_webhook.go]: yes, updated

Comment (nojnhuh) [reply api/v1beta1/azuremanagedmachinepool_webhook.go]: Yes to the SDK, not the CAPZ API

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**